group-coordinator/src/test/java/org/apache/kafka/coordinator/group/assignor/RangeAssignorTest.java

```diff
@@ -0,0 +1,574 @@
+        mkAssignment(expectedAssignment, topic1Uuid, Arrays.asList(0, 1));
+        mkAssignment(expectedAssignment, topic1Uuid, Collections.singleton(2));
+        // Topic 2 Partitions Assignment
+        mkAssignment(expectedAssignment, topic2Uuid, Arrays.asList(0, 1));
+        mkAssignment(expectedAssignment, topic2Uuid, Collections.singleton(2));
+        // Topic 3 Partitions Assignment
+        mkAssignment(expectedAssignment, topic3Uuid, Arrays.asList(0, 1));
+
+        // Test for stickiness
+        assertTrue(computedAssignment.members().get(consumerC)
+            .targetPartitions().get(topic2Uuid).contains(2),
+            "Stickiness test failed for Consumer C");
+        assertTrue(computedAssignment.members().get(consumerA)
+            .targetPartitions().get(topic1Uuid).containsAll(Arrays.asList(0, 1)),
+            "Stickiness test failed for Consumer A");
+
+        assertAssignment(expectedAssignment, computedAssignment);
+    }
+
+    private void mkAssignment(Map<Uuid, Set<Set<Integer>>> expectedAssignment, Uuid topicId, Collection<Integer> partitions) {
+        expectedAssignment.computeIfAbsent(topicId, k -> new HashSet<>()).add(new HashSet<>(partitions));
+    }
+
+    // We have a set of sets with the partitions that should be distributed amongst the consumers, if it exists then remove it from the set.
+    // The test is done like this since the order in which members are assigned partitions isn't guaranteed. We are just testing if the computed
+    // assignment contains the expected set of partitions, irrespective of which member got them.
+    private void assertAssignment(Map<Uuid, Set<Set<Integer>>> expectedAssignment, GroupAssignment computedGroupAssignment) {
+        for (MemberAssignment member : computedGroupAssignment.members().values()) {
+            Map<Uuid, Set<Integer>> computedAssignmentForMember = member.targetPartitions();
+            for (Map.Entry<Uuid, Set<Integer>> assignmentForTopic : computedAssignmentForMember.entrySet()) {
+                Uuid topicId = assignmentForTopic.getKey();
+                Set<Integer> assignmentPartitionsSet = assignmentForTopic.getValue();
+                assertTrue(expectedAssignment.get(topicId).contains(assignmentPartitionsSet));
+                expectedAssignment.remove(assignmentPartitionsSet);
+            }
+        }
+    }
+
+    private void assertCoPartitionJoinProperty(GroupAssignment groupAssignment) {
+        for (MemberAssignment member : groupAssignment.members().values()) {
+            Map<Uuid, Set<Integer>> computedAssignmentForMember = member.targetPartitions();
+            Set<Integer> compareSet = new HashSet<>();
+            for (Set<Integer> partitionsForTopicSet : computedAssignmentForMember.values()) {
+                if (compareSet.isEmpty()) {
+                    compareSet = partitionsForTopicSet;
+                }
+                assertEquals(compareSet, partitionsForTopicSet);
+            }
+        }
+    }
+}
```

Comment:
Should we add tests where we remove or add more than one members?

Reply:
changing multiple subscriptions has similar effects as adding and removing consumers and that test exists so I didn't add another one.